java/java.lsp.server/src/org/netbeans/api/lsp/server/NbCodeLanguageClient.java

```diff
@@ -26,10 +26,18 @@
 import org.eclipse.lsp4j.jsonrpc.services.JsonRequest;
 import org.eclipse.lsp4j.jsonrpc.validation.NonNull;
 import org.eclipse.lsp4j.services.LanguageClient;
+import org.netbeans.modules.java.lsp.server.protocol.DecorationRenderOptions;
+import org.netbeans.modules.java.lsp.server.protocol.HtmlPageParams;
+import org.netbeans.modules.java.lsp.server.protocol.QuickPickItem;
+import org.netbeans.modules.java.lsp.server.protocol.SetTextEditorDecorationParams;
+import org.netbeans.modules.java.lsp.server.protocol.ShowInputBoxParams;
+import org.netbeans.modules.java.lsp.server.protocol.ShowQuickPickParams;
+import org.netbeans.modules.java.lsp.server.protocol.ShowStatusMessageParams;
+import org.netbeans.modules.java.lsp.server.protocol.TestProgressParams;
 
 /**
  * An extension to the standard LanguageClient that adds several messages missing
- * from the official LSP protocol.s
+ * from the official LSP protocol.
  * @author sdedic
  */
 public interface NbCodeLanguageClient extends LanguageClient {
```

Comment:
Was this class ever meant to be an API? It is an interface. How do you plan to evolve it in the future?

Reply:
It has to (somehow) - otherwise the code wouldn't be able to call things like `showInputBox`. Additional messages can be added as `default` methods throwing UnsupportedOpEx. There are two additional ways: `final` class with protocol messages that will delegate all methods to NbCodeLanguageClient (internal API). Or the same with the `Context` proposed below.

Reply:
Why not call `DialogDisplayer.getDefault().notifyFuture(new NotifyDescriptor.InputLine(...))` instead of `showInputBox`?

Reply:
Eh ;) bad example then. Let's pick a message that is NOT bridged from regular NetBeans APIs, e.g. `disposeTextEditorDecoration` or `notifyNodeChange`.

Reply:
If you want to expose whole LSP - then providing access to `LanguageClient` may be the right thing. If you want to expose it including NetBeans extensions, possibly exposing `NbCodeLanguageClient` is the right thing - but then the protocol has to become (relatively) stable API - a review of our adhoc messages may be desirable (that's what we haven't paid much attention to yet).